heartbeat/server.js

```diff
@@ -26,15 +26,16 @@ db.on('compaction.done', () => {
 
 const FLIC_THRESHOLD_MILLIS = 180*1000
 const HEARTBEAT_THRESHOLD_MILLIS = 60*1000
+const PING_THRESHOLD_MILLIS = 320*1000
```

Comment:
What do you think about a 3-minute threshold for this.

I went back through the logs since Mario changed the wifi config and I do see periods where the Ping was unsuccessful for up to 389 seconds and then recovers on its own. It looks like most of the time it recovers in less than 60 seconds, but there are still quite a few times when it's up in the 100 or 200 second range before it self-recovers.

I want to balance the terrible-ness of it being down vs how long it would take one of us to get to a computer to manually fix it in any case.

On the other hand, knowing that these numbers can be so high, maybe we want to go back to the idea of the brute-force auto-restart after a much lower threshold that would take the system down but would at least show the red light to the Resident to let them know that it didn't work.

Reply:
I think another thing to consider is that a high threshold like 6 minutes in this case makes it likelier that we don't tune out the alerts and would probably signal truly exceptional/prolonged downtime cases?

Then there's the separate issue of tracking and working on eliminating the sub-threshold outages? I don't know if humans intervening on a command line somewhere are the right solution to those

Reply:
I'm not convinced that these short-duration episodes where the hub doesn't respond to pings actually represent real downtime. I'd want to see evidence of that before we start aggressively restarting the flic hub, especially since that would introduce a known amount of downtime while the flic hub reboots.

I think if it's commonly reaching ~200 seconds and then recovering on its own we should set the threshold so that these events don't trigger alerts. I see that you've got it set at 320 seconds or ~5 mins, which seems appropriate.

Reply:
That's a good point about us not knowing whether the short-duration episodes cause a real outage. We know that the long ones did.

Unfortunately to try it out we would need to be really on the ball and hope that someone at the Buchan is able to push a test button before it resolves itself. And even then, if the button worked and the last seen ping resolved itself at the same time, I wouldn't trust that we weren't just "lucky" that it resolved itself just when we wanted to press it.

As for tracking, these logs help but they aren't easy to parse out the valuable information. We could start logging to the DB now that we have one. But that probably isn't worth it unless we rule out the possibility of using the Flic Hub SDK.

So what I'm hearing from you both is that this looks fine as is for the time being. I think so too.